src/app_model/types/_keys/_keybindings.py

```diff
@@ -117,6 +169,33 @@ def to_int(self, os: Optional[OperatingSystem] = None) -> int:
             mods |= KeyMod.CtrlCmd if os.is_mac else KeyMod.WinCtrl
         return mods | (self.key or 0)
 
+    def _kb2mods(self) -> List[str]:
+        """Extract list of modifiers from this SimpleKeyBinding."""
+        mods = []
+        if self.ctrl:
+            mods.append("Ctrl")
+        if self.shift:
+            mods.append("Shift")
+        if self.alt:
+            mods.append("Alt")
+        if self.meta:
+            mods.append("Meta")
+        return mods
+
+    def to_text(
+        self, os: Optional[OperatingSystem] = None, use_symbols: bool = False
+    ) -> str:
+        """Get a user-facing string representation of this SimpleKeyBinding.
+
+        Optionally, the string representation can be constructed with symbols
+        like ↵ for Enter or OS specific ones like ⌘ for Cmd on MacOS.
+        """
+        joinchar, key_symbols, key_names = get_keys_user_facing_representation(os=os)
+        return joinchar.join(
+            key_symbols.get(x, x) if use_symbols else key_names.get(x, x)
```

Comment:
i wonder if much of this PR (including the look into the mapping here with `.get(x, x)`) should actually be a method on `KeyCode`.  Currently, this method is first casting `self.key` (which is a `KeyCode`) to a string, and then using that string to again index into another map.  What if did that more directly:

```python
class KeyCode(IntEnum):
    ...
    def to_os_symbol(self, os: Optional[OperatingSystem]) -> str: ...
    def to_os_name(self, os: Optional[OperatingSystem]) -> str: ...
```

and then put these lookups at a lower level in `_key_codes.py`?

that would also have the benefit of being able to directly get these symbols at a lower level, given something like `KeyCode.Meta` ... without first having to put it into a keybinding object

Reply:
Oh I see, I think that makes sense, will give it a try :+1: also thank you so much for the early feedback!

Reply:
Thank you! And actually, checking `_key_codes.py`, I'm wondering if it could make sense to add the symbols and names mappings to the logic over `_build_maps` along side the addition there of some new functions like:

```python
def _keycode_to_os_symbol(keystr: str, os: OperatingSystem) -> str:...
def _keycode_to_os_name(keystr: str, os: OperatingSystem) -> str:...
```

Seems like that could be a good place 🤔 What do you think?

Reply:
yeah i agree, `_build_maps` is the right place to construct these lookups.

Reply:
Pushed some changes which hopefully should improve code organization :+1: Also, checking https://github.com/napari/napari/pull/5064 I think another functionality that could be added is https://github.com/napari/napari/blob/main/napari/utils/interactions.py#L312 so not completly 1 to 1 but kind of a `from_text` method. Does that makes sense?

Reply:
i believe #213 should take care of that?  but let me know if I'm not understanding what that code does

Reply:
Give that a check and I think the work at #213 partialy captures the `parse_platform` logic at  https://github.com/napari/napari/blob/main/napari/utils/interactions.py#L312 The missing thing I would say is also being able to parse symbols from non modifier keys like enter, arrows, backspace, delete, tab, space, etc. Could it be worthy to further extend the parsing capabilities to any symbol or representation that `os_symbol` and `os_name` can return? From a quick check, maybe extending the `KEYCODE_FROM_LOWERCASE_STRING` dictionary could be a way to achieve this? So something like:

```python
KEYCODE_FROM_LOWERCASE_STRING: Dict[str, KeyCode] = {
    # two special cases for assigning os-specific strings to the meta key
    'win': KeyCode.Meta,
    'cmd': KeyCode.Meta,
    # special cases for symbols and other os-specific strings
    '↵': KeyCode.Enter,
    '→': KeyCode.RightArrow,
    ...
}
```

Reply:
ah I see... yeah, perhaps.  Lets discuss that in another issue though.  Seems like it might be a cool but YAGNI feature?  do you actively parse strings with key symbols in napari?  or was it more of a "why not allow this" sort of feature?

Reply:
Created https://github.com/pyapp-kit/app-model/issues/214 :+1: and I think that was added to be able to store and comper shortcuts in a canonical/normalized form from a representation of a keybinding like the one you would get from the GUI (like for example the napari shortcuts widget) 🤔 So, to support logic like the following: 

https://github.com/napari/napari/blob/5e685a20c9563a42576ead39a4f0286337a52ce6/napari/_qt/widgets/qt_keyboard_settings.py#L403-L428

Reply:
Ok, plenty to discuss there.  I would definitely keep things as KeyCodes and not strings as much as possible.  So, rather than using current.text() in that widget, you should take advantage of itemData(), or somehow have the visual representation of the keybinding in the widget not _be_ the actual data that napari uses in the model.  (Think of this as a classic model view thing, where the model is the KeyCode, and the view is some string that the user sees)

Reply:
Yep, that makes sense to me :+1: I think I will mark this as ready for review then. Let me know if further changes are needed here!